src/Swarm/Game/Step.hs

```diff
@@ -68,26 +68,19 @@ evalStepsPerTick = 100
 --   from a file when the whole program starts up).
 gameTick :: (MonadState GameState m, MonadIO m) => m ()
 gameTick = do
-  -- Note, it is tempting to do the below in one line with some clever
-  -- lens combinator, but it's not possible.  We want to do an
-  -- effectful traversal over a piece of the state (i.e. step each
-  -- robot in the robot map, where stepping a robot could have effects
-  -- on the game state), but the problem is the effects could in
-  -- theory include modifying the very state we are traversing over.
-  -- In fact, stepping one robot can even have effects on other robots
-  -- (e.g. if one robot gives an item to another), so we can't even
-  -- use something like M.traverseMaybeWithKey --- we have to get the
-  -- names of all robots and then step them one at a time.
-  robotNames <- uses robotMap M.keys
+  wakeUpRobotsDoneSleeping
+  robotNames <- use activeRobots
+  time <- use ticks
   forM_ robotNames $ \rn -> do
     mr <- uses robotMap (M.lookup rn)
     case mr of
       Nothing -> return ()
       Just curRobot -> do
-        curRobot' <- tickRobot curRobot
+        curRobot' <- tickRobot time curRobot
```

Comment:
Why do we thread the `time` as a parameter through all these functions that have access to the `GameState`?  Can't we just move the `use ticks` to the `stepCEK` function where it is actually needed?

Reply:
Good point, I didn't connect the dots and realize that the time was available in the ambient monad, will fix.